### DISCUSSION TITLE
How to use class_labels in the Unet2DConditionalModel or Unet2DModel when forward?

### DISCUSSION THREAD

**Discussion by boqian-li** [huggingface/diffusers#6887] [Q&A]
Hi, I want to know what the shape or format of "class" is if I want to add the class condition to the unet? Just set the classe_labels 0, 1, 2, 3?

Unet2DModel: class_labels (torch.FloatTensor, optional, defaults to None) — Optional class labels for conditioning. Their embeddings will be summed with the timestep embeddings.

Unet2DConditionalModel: class_labels (torch.Tensor, optional, defaults to None) — Optional class labels for conditioning. Their embeddings will be summed with the timestep embeddings. timestep_cond — (torch.Tensor, optional, defaults to None): Conditional embeddings for timestep. If provided, the embeddings will be summed with the samples passed through the self.time_embedding layer to obtain the timestep embeddings.

**Comment by sayakpaul** [comment]
Cc: @kashif @dg845 in case they know an example usage.

**Reply by boqian-li** [reply]
Hi, is there any update on this?

**Comment by boqian-li** [comment]
Any update or good examples?
